rest_framework_swagger/views.py

```diff
@@ -109,6 +116,13 @@ def get(self, request):
             }),
         })
 
+    def get_base_path(self):
+        protocol = SWAGGER_SETTINGS.get('protocol', 'http')
```

Comment:
This patch is breaking backward compatibility because `request.is_secure()` is ignored and `http` is always used unless I add `{'protocol': 'https'}` in my settings.

`request.is_secure()` should be the only reliable and consistent way to go here.

I do not understand why adding `protocol` in swagger settings is a good idea?
You want to use `http` even if `request.is_secure() is True` ?
If `nginx` is forced to connect with `http` and if `request.is_secure()` return `True`, it means you probably have something misconfigured in your settings.